internal/go_repository.bzl

```diff
@@ -28,11 +53,13 @@ def _go_repository_impl(ctx):
         for key in ("commit", "tag", "vcs", "remote", "version", "sum", "replace"):
             if getattr(ctx.attr, key):
                 fail("cannot specifiy both urls and %s" % key, key)
+        auth = _get_auth(ctx, ctx.attr.urls)
         ctx.download_and_extract(
             url = ctx.attr.urls,
             sha256 = ctx.attr.sha256,
             stripPrefix = ctx.attr.strip_prefix,
             type = ctx.attr.type,
+            auth = auth,
```

Comment:
It looks like this will only work for HTTP mode. It also needs to work for module mode though (the `ctx.attr.version` branch).

I think that just means making sure `NETRC` is set to the correct file in `env_keys`, but please verify.